momentGW/pbc/ints.py

```diff
@@ -307,6 +312,82 @@ def transform(self, do_Lpq=None, do_Lpx=True, do_Lia=True):
 
         logger.timer(self, "transform", *cput0)
 
+    def input_integrals(self):
+        """
+        Build CDERI elements using THC integrals imported from a h5py
+        file containing a dictionary of components. Inside the dict, a
+        'collocation_matrix' and a 'coulomb_matrix' must be contained
+        with shapes (MO, aux) and (aux,aux) respectively.
+        """
+        if self.input_path is None:
+            raise ValueError(
+                "A file path containing the THC integrals is needed for the THC implementation"
+            )
+        if "thc" not in self.input_path or "THC" not in self.input_path:
+            raise ValueError("File path must contain 'thc' or 'THC' for THC implementation")
+
+        thc_eri = h5py.File(self.input_path, "r")
+
+        kpts_imp = np.array(thc_eri["kpts"])
+
+        if kpts_imp.shape[0] != len(self.kpts):
+            raise ValueError("Number of kpts imported differs from pyscf")
+        if not np.allclose(kpts_imp, self.kpts._kpts) and not np.allclose(
+            kpts_imp, -self.kpts._kpts
+        ):
+            raise ValueError("Different kpts imported to those in pyscf")
+
+        Lpx = {}
+        Lia = {}
+        Lai = {}
+        self._naux = [np.array(thc_eri["coulomb_matrix"])[0, ..., 0].shape[0]] * len(self.kpts)
+        for q in self.kpts.loop(1):
+            for ki in self.kpts.loop(1, mpi=True):
+                kj = self.kpts.member(self.kpts.wrap_around(self.kpts[q] + self.kpts[ki]))
+
+                Lpx_k = np.zeros((self.naux[q], self.nmo, self.nmo_g[kj]), dtype=complex)
+                Lia_k = np.zeros((self.naux[q], self.nocc_w[ki] * self.nvir_w[kj]), dtype=complex)
+                Lai_k = np.zeros((self.naux[q], self.nocc_w[ki] * self.nvir_w[kj]), dtype=complex)
+
+                cou = np.array(thc_eri["coulomb_matrix"])[q, ..., 0]
```

Comment:
just to check, these are `h5py` datasets right? I would maybe prefer `np.asarray` just stylistically. `np.asarray` says "copy me if i'm not already a numpy array" whereas `np.array` says "copy me into a new numpy array"